Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -84,9 +84,9 @@ public function transform($entityOrCollection)
             return $result;
         }
         if ($this->multiple) {
-            if (substr(get_class($entityOrCollection), -1 * strlen($this->className)) == $this->className) {
+            if (!is_array($entityOrCollection) && substr(get_class($entityOrCollection), -1 * strlen($this->className)) == $this->className) {
```

Comment:
The `is_array` check should be the second condition in the `if` statement to improve the performance.